airflow/providers/google/cloud/hooks/compute_ssh.py

```diff
@@ -191,6 +197,65 @@ def intify(key, value, default):
             if self.cmd_timeout is NOTSET:
                 self.cmd_timeout = CMD_TIMEOUT
 
+    def _check_running_in_parallel(self) -> bool:
+        """
+        Helper function to determine if there are several tasks that are running in parallel.
+
+        :return: boolean, True if there are more than 1 task running at the same time
+        """
+        dag_id = self.context["dag"].dag_id  # type: ignore
+        running_tasks = []
+        with create_session() as session:
+            for task_instance in (
+                session.query(TaskInstance)
+                .filter(TaskInstance.state == State.RUNNING, TaskInstance.dag_id == dag_id)
+                .all()
+            ):
+                running_tasks.append(task_instance)
+        return True if len(running_tasks) > 1 else False
+
+    def _create_bucket_for_lock(self, bucket_name: str) -> None:
+        """
+        Helper function to create a bucket to hold file lock.
+
+        :param bucket_name: The name for the bucket to be created
+        """
+        hook = GCSHook(gcp_conn_id="google_cloud_default")
+        try:
+            hook.create_bucket(bucket_name=bucket_name, location="US", project_id=self.project_id)
+        except Conflict:  # HTTP 409
+            self.log.info("Bucket %s already exists", bucket_name)
+
+    def _acquire_lock(self, lock_path: str) -> bool:
+        """
+        Creates a lock with the specified GCS path.
+        :param lock_path: the lock's GCS path
+        :return: True if the lock was acquired.
+        """
+        echo = Popen('echo "lock"', shell=True, stdout=PIPE)
+        try:
+            # command to upload the content of echo command to the specified file in gcs.
+            # Specified flags for the command will ensure that the upload only occurres if the generation
+            # of the file is 0, that indicate that the file was not modified after the creation to
+            # prevent unintentional overwriting of file
+            cmd = f'gsutil -q -h "x-goog-if-generation-match:0" cp - {lock_path}'
+
+            # executing command above, passing output of echo command as an input param to the cmd
+            check_output(cmd, stdin=echo.stdout, shell=True, stderr=STDOUT)
+            echo.wait()
+            return True
+        except CalledProcessError:
+            return False
+
+    def _release_lock(self, lock_path: str) -> None:
+        """
+        Releases the specified lock.
+        :param lock_path: the lock's GCS path with the gs://bucket-name/file-name format
+        :return: None.
+        """
+        cmd = f"gsutil -q rm {lock_path}"
+        check_output(cmd, shell=True, stderr=STDOUT)
```

Comment:
I think it would be good to cover it with try/except because if we fail to release the lock, then all the processes will be blocked forever. Perhaps it is worth thinking about negative scenarios that could lead to this failure and handle them, e.g. network issues? WDYT?